### PR TITLE
feature: bscp增加前置、后置脚本功能

### PR DESCRIPTION
**Pull request by fireyun**
### 增加如下脚本API
| API说明      | API名称     | API Path                                     | Method |
| ------------ | ----------- | -------------------------------------------- | ------ |
| 创建脚本     | create_hook | /api/v1/config/apps/{app_id}/hooks           | POST   |
| 获取脚本列表 | list_hooks  | /api/v1/config/apps/{app_id}/hooks/list      | POST   |
| 更新脚本     | update_hook | /api/v1/config/apps/{app_id}/hooks/{hook_id} | PUT    |
| 删除脚本     | delete_hook | /api/v1/config/apps/{app_id}/hooks/{hook_id} | DELETE |

close issue #1987

### CODE REVIEW

**Comment by tencentblueking-adm** [comment]
[CLA assistant check](https://bk-cla.bktencent.com:443/TencentBlueKing/bk-bcs?pullRequest=2034) <br/>All committers have signed the CLA.

File: bcs-services/bcs-bscp/pkg/dal/dao/audit_builder.go [hunk 1]
```diff
@@ -517,6 +560,19 @@ func (ab *AuditBuilder) getStrategy(strategyID uint32) (*table.Strategy, error)
 	return one, nil
 }
 
+func (ab *AuditBuilder) getHook(hookID uint32) (*table.Hook, error) {
+	filter := fmt.Sprintf(`SELECT %s FROM %s WHERE id = %d AND biz_id = %d`,
```
Comment (ifooth) [inline bcs-services/bcs-bscp/pkg/dal/dao/audit_builder.go]: 这个可以参考https://github.com/TencentBlueKing/bk-bcs/pull/2028改下，防止sql注入问题

File: bcs-services/bcs-bscp/pkg/dal/dao/hook.go [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+
+	// generate a hook id and update to hook.
+	id, err := dao.idGen.One(kit, table.HookTable)
+	if err != nil {
+		return 0, err
+	}
+
+	g.ID = id
+
+	sql := fmt.Sprintf(`INSERT INTO %s (%s)	VALUES(%s)`, table.HookTable,
```
Comment (ifooth) [inline bcs-services/bcs-bscp/pkg/dal/dao/hook.go]: 同上

File: bcs-services/bcs-bscp/pkg/dal/dao/audit_builder.go (hunk 1 above)
Comment (fireyun) [reply bcs-services/bcs-bscp/pkg/dal/dao/audit_builder.go]: 已调整

**Review by ifooth — APPROVED** [review]

**Pull request merged**